### PR TITLE
test: merge deterministic integration tests into one step

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Now we have 4 deterministic integration test steps:

image

They use the same compiled binary but running with different filters. I think this is not necessary, and let's put them together to save EC2 instances.

By the way, the `TEST_NUM` is unified.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: ci/workflows/main-cron.yml
Comment (BugenZhao) [inline ci/workflows/main-cron.yml:128]:
```suggestion
    command: "TEST_NUM=60 ci/scripts/deterministic-it-test.sh"
```

**Review by BugenZhao — COMMENTED** [review]
> 4 deterministic integration test steps

This was to make them run in parallel to reduce the total running time. While it seems they won't be a bottleneck even running sequentially now, merging together LGTM.

File: ci/workflows/main.yml
Comment (kwannoel) [inline ci/workflows/main.yml:213]: Might TLE since it includes more tests now, but there's no way to tell, until after merging.

**Review by kwannoel — APPROVED** [review]
LGTM after bugen's comment is fixed.

**Comment by kwannoel** [comment]
Somethiing else to take note of is that long running jobs can get cancelled in the middle of their run. Re-running them will take more time.

File: ci/workflows/main.yml
Comment (fuyufjh) [reply ci/workflows/main.yml:213]: Yeah I will keep an eye on the workflow

**Comment by fuyufjh** [comment]
> Somethiing else to take note of is that long running steps (1h+) can get cancelled in the middle of their run. Re-running them will take more time. Compared to retrying a single step in the workflow.

Yes. If the main or main-cron workflow fails (they are using TEST_NUM = 30/60), I'll revert and separate them as status quo

**Pull request merged**
